backend/curriculum_tracking/tests/test_api_views.py

```diff
@@ -234,6 +238,51 @@ def test_set_project_link(self):
         project.refresh_from_db()
         self.assertEqual(project.link_submission, link_2)
 
+    @mock.patch.object(RecruitProject, "setup_repository")
+    def setup_project_repo_call_from_api_action_option(self, get_repo):
+
+        JAVASCRIPT = "js"
+        super_user = factories.UserFactory(is_superuser=True)
+
+        def get_repo_mock(
+            github_auth_login="", repo_full_name="", api=None, response404=None, add_collaborators=None
+        ):
+            return {
+                "full_name": f"me/{repo_full_name}",
+                "owner": {"login": "me"},
+                "ssh_url": f"https://{repo_full_name}.git",
+                "private": True,
+                "created_at": timezone.now().strftime(GITHUB_DATETIME_FORMAT),
+                "archived": False,
+            }
+        get_repo.side_effect = get_repo_mock
+
+        content_item = factories.ContentItemFactory(
+            content_type=ContentItem.PROJECT,
+            project_submission_type=ContentItem.REPOSITORY,
+            flavours=[JAVASCRIPT],
+        )
+
+        card = factories.AgileCardFactory(
+            content_item=content_item,
+            flavours=[JAVASCRIPT],
+            status=AgileCard.READY,
+            recruit_project=None,
+        )
+
+        assignee = SocialProfileFactory().user
+        card.assignees.set([assignee])
+        card.start_project()
```

Comment:
This line also calls setup_repository. It might be good to make sure that setup_repository has been called the correct number of times.

Reply:
This dramatiq thing is strange to me, if I for instance put a breakpoint in the view and then run the test, once I skip past this part "response = recruit_project_setup_repository.send(card.recruit_project.id)" of the view and then print response it shows this:

Message(queue_name='default', actor_name='recruit_project_setup_repository', args=(1,), kwargs={}, options={}, message_id='41f05619-7664-4701-90d4-0fb4074b403a', message_timestamp=1639572949391)

which means it worked, however, in the test itself when I get to this "self.assertTrue(project.setup_repository.called)" it will return False saying setup_repository was never called which is not the case when you consider that the view tells you the message was sent to an actor called 'recruit_project_setup_repository' with the correct arguments. 

In fact, if I change the view so that I no longer make use of dramatiq then everything works perfectly, even the test, but the moment I use dramatiq it says setup_repository wasn't called, is there not another way to test dramatiq actors?

Reply:
Finally got the test to work with dramatiq.